__tests__/__unit__/dataset/actions.unit.test.ts

```diff
@@ -0,0 +1,1508 @@
+        createGlobalMocks();
+        blockMocks = createBlockMocks();
+        const parent = new ZoweDatasetNode("parent", vscode.TreeItemCollapsibleState.Collapsed,
+            blockMocks.datasetSessionNode, blockMocks.session);
+
+        mocked(vscode.window.showInputBox).mockResolvedValue("");
+        await dsActions.createMember(parent, blockMocks.testDatasetTree);
+
+        expect(mocked(zowe.Upload.bufferToDataSet)).not.toBeCalled();
+    });
+    it("Checking of member creation for favorite dataset", async () => {
+        createGlobalMocks();
+        blockMocks = createBlockMocks();
+        const parent = new ZoweDatasetNode("parent", vscode.TreeItemCollapsibleState.Collapsed,
+            blockMocks.datasetSessionNode, blockMocks.session);
+        const nonFavoriteLabel = parent.label;
+        parent.label = `[${blockMocks.datasetSessionNode.label}]: ${parent.label}`;
+        parent.contextValue = globals.DS_PDS_CONTEXT + globals.FAV_SUFFIX;
+
+        mocked(vscode.window.showInputBox).mockResolvedValue("testMember");
+        mocked(vscode.window.withProgress).mockImplementation((progLocation, callback) => {
+            return callback();
+        });
+        jest.spyOn(blockMocks.mvsApi, "getContents").mockResolvedValueOnce({
+            success: true,
+            commandResponse: null,
+            apiResponse: {
+                etag: "123"
+            }
+        });
+
+        await dsActions.createMember(parent, blockMocks.testDatasetTree);
+
+        expect(mocked(vscode.window.showInputBox)).toBeCalledWith({ placeHolder: "Name of Member" });
+        expect(mocked(zowe.Upload.bufferToDataSet)).toBeCalledWith(
+            blockMocks.zosmfSession,
+            Buffer.from(""),
+            nonFavoriteLabel + "(testMember)",
+            undefined
+        );
+    });
+});
+
+describe("Dataset Actions Unit Tests - Function refreshPS", () => {
+    let blockMocks;
+
+    function createBlockMocks() {
+        const session = generateISession();
+        const imperativeProfile = generateIProfile();
+        const zosmfSession = generateBasicZosmfSession(imperativeProfile);
+        const treeView = generateTreeView();
+        const datasetSessionNode = generateDatasetSessionNode(session, imperativeProfile);
+        const mvsApi = generateMvsApi(imperativeProfile);
+        bindMvsApi(mvsApi);
+
+        return {
+            session,
+            zosmfSession,
+            treeView,
+            imperativeProfile,
+            datasetSessionNode,
+            mvsApi,
+            testDatasetTree: generateDatasetTree(datasetSessionNode, treeView)
+        };
+    }
+
+    afterAll(() => jest.restoreAllMocks());
+
+    it("Checking common PS dataset refresh", async () => {
+        globals.defineGlobals("");
+        createGlobalMocks();
+        blockMocks = createBlockMocks();
+        const node = new ZoweDatasetNode("HLQ.TEST.AFILE7", vscode.TreeItemCollapsibleState.None, blockMocks.datasetSessionNode, null);
```

Comment:
The same node is used in almost all it() blocks, can it be declared inside createBlockMocks()?